src/model/resources/fusion/fusion_policy_adjustments.jl

```diff
@@ -0,0 +1,141 @@
+    FUSION = ids_with(gen, fusion)
+    applicable_resources = intersect(FUSION, THERM_COMMIT)
+
+    resource_component(y) = resource_name(gen[y])
+
+    for y in applicable_resources
+        _fusion_capacity_reserve_margin_adjustment!(EP, inputs, resource_component(y), y)
+    end
+end
+
+# inner-loop function: loops over Capacity Reserve Margin zones, for one resource
+# and actually adjusts the eCapResMarBalance expression
+function _fusion_capacity_reserve_margin_adjustment!(EP::Model,
+        inputs::Dict,
+        resource_component,
+        y::Int)
+    T = inputs["T"]
+    timesteps = collect(1:T)
+    ncapres = inputs["NCapacityReserveMargin"]
+
+    eCapResMarBalance = EP[:eCapResMarBalance]
+
+    for capres_zone in 1:ncapres
+        adjustment = fusion_capacity_reserve_margin_adjustment(
+            EP, inputs, resource_component, y, capres_zone, timesteps)
+        add_similar_to_expression!(eCapResMarBalance[capres_zone, :], adjustment)
+    end
+    return
+end
+
+# Get the amount for one resource component in one CRM zone
+function fusion_capacity_reserve_margin_adjustment(EP::Model,
+        inputs::Dict,
+        resource_component::AbstractString,
+        y::Int,
+        capres_zone::Int,
+        timesteps::Vector{Int})
+    gen = inputs["RESOURCES"]
+    component = gen[y]
+    eTotalCap = EP[:eTotalCap][y]
+
+    capresfactor = derating_factor(component, tag=capres_zone)
+    if capresfactor == 0.0
+        return AffExpr.(zero.(timesteps))
+    end
+    dwell_time = Float64(get(component, :dwell_time, 0.0))
+    component_size = get(component, :cap_size, 0.0)
+
+    from_model(f::Function) = EP[f(resource_component)]
+
+    ePassive = from_model(fusion_parasitic_passive_name)
+    eActive = from_model(fusion_parasitic_active_name)
+    eStartPower = from_model(fusion_pulse_start_power_name)
+    ePulseStart = component_size * from_model(fusion_pulse_start_name)
+    ePulseUnderway = component_size * from_model(fusion_pulse_underway_name)
+
+    capacity_to_underway_adj = capresfactor * (ePulseUnderway[timesteps] .- eTotalCap)
+    parasitic_adj = _fusion_crm_parasitic_adjustment.(
+        capresfactor, ePassive[timesteps], eActive[timesteps], eStartPower[timesteps])
+    dwell_adj = -capresfactor *
+                _fusion_dwell_avoided_operation.(dwell_time, ePulseStart[timesteps])
+
+    total_adj = @expression(EP, [t in 1:T], capacity_to_underway_adj[t] + parasitic_adj[t] + dwell_adj[t])
+
+    # Cancel out the dependence on down_var, since CRM is related to power, not capacity
+    if y in ids_with_maintenance(gen)
+        maint_adj = thermal_maintenance_and_fusion_capacity_reserve_margin_adjustment(
+            EP, inputs, y, capres_zone, timesteps)
+        add_to_expression!.(total_adj, maint_adj)
+    end
+
+    return total_adj
+end
+
+# alias for better parallelism in effective_capacity.jl
+thermal_fusion_capacity_reserve_margin_adjustment = fusion_capacity_reserve_margin_adjustment
```

Comment:
```suggestion
const thermal_fusion_capacity_reserve_margin_adjustment = fusion_capacity_reserve_margin_adjustment
```
to improve performance

Reply:
done.